io/CMakeLists.txt

```diff
@@ -31,9 +31,15 @@ set_target_properties(io PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BIN
 target_compile_definitions(io PRIVATE "${_io_defs_private}")
 target_include_directories(io PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/mod>)
 
-target_link_libraries(io PRIVATE fms
-                                 gfsphysics
-                                 ccppdriver)
+if(32BIT)
+   target_link_libraries(io PRIVATE FMS::fms_r4
+                                    gfsphysics
+                                    ccppdriver)
+else()
+   target_link_libraries(io PRIVATE FMS::fms_r8
+                                    gfsphysics
+                                    ccppdriver)
+endif()
```

Comment:
This is fine for now, but in the subsequent PR's we could use an alias target when FMS is provided as a module or "third-party library" or built in the UFS (as is the case now).
This can be achieved by something like this  in the UFS:
```
find_package(FMS QUIET)
if (FMS_FOUND)
  if (32BIT)
    add_library(fms ALIAS FMS::fms_r4)
  else()
    add_library(fms ALIAS FMS::fms_r8)
  endif()
else()
  add_subdirectory(FMS)
  if (32BIT)
    add_library(fms ALIAS FMS::fms_r4)
  else()
    add_library(fms ALIAS FMS::fms_r8)
  endif()
endif()
```

All downstream targets can then use `fms` without the need for and `if-else` block scattered everywhere.